compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/InitMethodGen.java

```diff
@@ -187,18 +189,23 @@ public void generateModuleInitializer(ClassWriter cw, BIRNode.BIRPackage module,
         MethodGenUtils.visitReturn(mv);
     }
 
-    public void enrichPkgWithInitializers(Map<String, JavaClass> jvmClassMap, String typeOwnerClass,
+    public void enrichPkgWithInitializers(JvmPackageGen jvmPackageGen, Map<String, BIRFunctionWrapper> birFunctionMap,
+                                          Map<String, JavaClass> jvmClassMap, String typeOwnerClass,
                                           BIRNode.BIRPackage pkg, List<PackageID> moduleImports) {
         JavaClass javaClass = jvmClassMap.get(typeOwnerClass);
         BIRNode.BIRFunction initFunc = generateDefaultFunction(moduleImports, pkg, MODULE_INIT_METHOD,
                                                                MethodGenUtils.INIT_FUNCTION_SUFFIX);
         javaClass.functions.add(initFunc);
         pkg.functions.add(initFunc);
+        birFunctionMap.put(JvmCodeGenUtil.getPackageName(pkg.packageID) + MODULE_INIT_METHOD,
```

Comment:
Why are we adding the module functions to the `birFunctionMap`? Are we including these to the `call()` method as well?

Reply:
Yes. We need this change since call method use birFucntionMap to get function wrapper when iterating through functions and it will become null. Other way is to specifically handle init and start but I think we can add those to call method if needed to call them in the future.